orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/manifest/DeployManifestStage.java

```diff
@@ -43,4 +59,25 @@ public void taskGraph(Stage stage, TaskNode.Builder builder) {
         .withTask(CleanupArtifactsTask.TASK_NAME, CleanupArtifactsTask.class)
         .withTask(BindProducedArtifactsTask.TASK_NAME, BindProducedArtifactsTask.class);
   }
+
+  public void afterStages(@Nonnull Stage stage, @Nonnull StageGraphBuilder graph) {
+    DeployManifestContext context = stage.mapTo(DeployManifestContext.class);
+    TrafficManagement trafficManagement = context.getTrafficManagement();
+    if (trafficManagement != null &&
+        trafficManagement.isEnabled() &&
+        trafficManagement.getOptions() != null &&
+        trafficManagement.getOptions().getStrategy() != null) {
```

Comment:
I wonder if you could simplify this by doing 
```
ManifestStrategyType strategy = Optional.of(trafficManagement).map(TrafficManagement::getOptions).map(TrafficManagement::getStrategy).orElse(null);

if (strategy != null) {
...
```

Reply:
that would be nice to pair with a guard, something like:
```
ManifestStrategyType strategy = Optional.of(trafficManagement).map(TrafficManagement::getOptions).map(TrafficManagement.Options::getStrategy).orElse(null);

if (strategy == null) { 
  return
}
...
```

Reply:
Thanks for the fancy Java tips, updated!